### ISSUE TITLE
Allow to use resource/body object directly

### ISSUE THREAD

**Issue by Maxim-Mazurok** [Maxim-Mazurok/google-api-typings-generator#364]
Instead of this:
```js
await gapi.client.analyticsreporting.reports.batchGet({
  resource: {
    reportRequests: [/*...*/],
  },
});
```
we want to use this:
```js
await gapi.client.analyticsreporting.reports.batchGet({
  reportRequests: [/*...*/],
});
```

First, we need to investigate if this is consistent across different APIs.
For example, [`body` approach](https://github.com/Maxim-Mazurok/google-api-typings-generator/tree/master#resource-vs-body) worked for Sheets API, but not for Analytics Reporting API.

**Comment by hakimio** [comment]
Not sure why directly specifying `reportRequests` for Analytics Reporting API doesn't work for you, because it works perfectly fine for me.

**Comment by Maxim-Mazurok** [comment]
You're right, direct approach also works fine for me. It's just that our current types definition does not support this direct approach.
So, this issue was created so that type definitions will be changed to allow this (totally valid and working) direct approach.
